tests/ltp/patches/ltp_listxattr_listxattr01_fix.patch

```diff
@@ -0,0 +1,49 @@
++ Patch Description: Tests were failing as no xattr support in loop device
++ So modofied the tests to use root filesystem with xattr support
+diff --git a/testcases/kernel/syscalls/listxattr/listxattr01.c b/testcases/kernel/syscalls/listxattr/listxattr01.c
+index 62198b2a7..ca3e70232 100644
+--- a/testcases/kernel/syscalls/listxattr/listxattr01.c
++++ b/testcases/kernel/syscalls/listxattr/listxattr01.c
+@@ -31,7 +31,9 @@
+ #define VALUE  "test"
+ #define VALUE_SIZE     (sizeof(VALUE) - 1)
+ #define KEY_SIZE    (sizeof(SECURITY_KEY1) - 1)
+-#define TESTFILE    "testfile"
++#define TESTFILE    "mntpoint/testfile"
++#define MNTPOINT    "mntpoint"
++#define DIR_MODE    (S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH)
+
+ static int has_attribute(const char *list, int llen, const char *attr)
+ {
+@@ -64,17 +66,29 @@ static void verify_listxattr(void)
+ }
+
+ static void setup(void)
+-{
++{
++       rmdir(MNTPOINT);
++       SAFE_MKDIR(MNTPOINT, DIR_MODE);
++       SAFE_MOUNT("/dev/vda", MNTPOINT, "ext4", 0, NULL);
```

Comment:
Can you explain what this line does? The root block device is already mounted.

Reply:
@prp actually this is a great opportunity to discuss about that and find the best solution for enablng this kind of tests. This approach is used for many test cases. @shaikshavali1 please you also get involved in this discussion.

@prp since we cannot use loop back device with sgx-lkl due to  memory limitations of sgx-lkl, here this mounting approach is used. We are mounting /dev/vda and running tests and then unmounting. This approah is used for 50+ enabled LTP tests. We want to understand if there is an issue with this approach and what is a better approach and we wll appriciate your expertiese and input here.

Here is Shaik's explanation about why we are using this approach:
```
Default size of LKL memory is set to 32M. This is kept low due to limited size of EPC (Enclave page cache) and to avoid page cache being swapped out.
 
In tests, where loop device is used ltp framework was causing a problem, as it tries to create a 256M in memory file system.
ltp framework perform below activity.
1. Create a .img of size 256M
2. Create a loop device and mount.
 
Also, it needs to run mkfs utility using unsupported system() system call. So, the work around on Peter's suggestion was to mount the root file system and test.
 
So, in some tests, where it mounts test fs, was replaced with root fs. eg: For attribute related tests the file system needs to be mounted with user_xattr option. But this was not set for "/". So, we had to mount root file system with the option and test, that's why we cannot use existing root filesystem and needed to mount /dev/vda with required options.
 
```

Reply:
@prp, The root fs is not mounted with extended attributes option. In this test case, a test file is created on current directory and try to set extended attribute on test file. While setting the attribute below error is observed. 
"safe_macros.c:877: CONF: listxattr01.c:70: no xattr support in fs or mounted without user_xattr option"

To resolve this issue "/dev/vda" is mounted explicitly to a newly created directory ./mntpoint. After this change the setxattr is able to set the extended attribute of the file.

Reply:
`/dev/vda` is the block device with the root file system in the enclave. It is already mounted and being used. I do not understand how you can mount it again.

Reply:
@prp, I will modify the test as per your suggestion and check the results.

Reply:
I modified the test case to create a test file  in "/" and "/ltp". In both the cases, the test case is passing.

Reply:
Does that mean that the root file system has xattr support? I thought we were mounting it without that option?